### ISSUE TITLE
Handling large media files

### ISSUE THREAD

**Issue by ghost** [ocaml/ood#58]
This topic has been mentioned in passing in previous PRs. Consider implementing a new server for image serving called image.ocaml.org using a system like thumbor. This will reduce the amount of data that need to be transferred between ood and v3, when v3 updates it's ood copy, and it will also allow for v3 to request scaled images on the fly.

**Comment by tmattio** [comment]
Thanks @kanishka-work!

I agree with the need to serve the image from a CDN, but I'm not sure I'm following why we'd need a server to scale the images. This seems like a lot of work for something that can be done with a little bit of CSS (or two tailwind classes). Am I missing something?

**Comment by ghost** [comment]
I think some websites scale the image on the server (at the moment of the request) to minimize the image download size for mobile users.

**Comment by patricoferris** [comment]
Yeah that seems reasonable to me -- CDNs like cloudinary let you do this on the fly using URL queries or something like that so you can specify what resolution, filters etc. (greatly under-selling all of things you can do https://cloudinary.com/cookbook). But if we rolled our own we could just create scaled versions and use `srcset` et al. to achieve the desired reduction in bytes going over the wire. And as @kanishka-work pointed out something like thumbor would give the subset we are interested in and we could host our own I think. 

This question goes beyond just images though as we are now discovering (see #48 where we discussed what to do with https://github.com/ocaml/ocaml.org-media), ood can't (shouldn't) do all of this so we might want something like a CDN + Git LFS or something... One to think about :))

**Comment by ghost** [comment]
Feel free to reword or close this issue, if you have a more concrete direction in mind. I created this issue to simply start the conversation.

**Comment by ghost** [comment]
Once this issue is complete, then we might consider using next/image together with a custom loader: https://nextjs.org/docs/api-reference/next/image#loader. The custom loader can be implemented as part of https://github.com/ocaml/v3.ocaml.org/issues/430.
